torch/testing/_internal/common_device_type.py

```diff
@@ -843,7 +843,20 @@ def _parametrize_test(self, test, generic_cls, device_cls):
                 try:
                     @wraps(test)
                     def test_wrapper(*args, **kwargs):
-                        return test(*args, **kwargs)
+                        try:
+                            return test(*args, **kwargs)
+                        except Exception as e:
+                            tracked_input = get_tracked_input()
+                            if tracked_input is not None:
+                                raise Exception(
+                                    f"Error caused by {tracked_input.input_type} "
+                                    f"{tracked_input.index}: {tracked_input.input_val}") from e
+                            raise e
```

Comment:
This is pretty funny, and I am OK with the stack-crawling approach we took here.

A feature request is: there should be some way to, given this TrackedInput information,  actually be able to reproduce the test locally. One idea for how to do this, taken from Hypothesis, is to have some sort of decorator that we could apply to a function:
```py
# On a local build, we can apply the following "example" decorator.
# It modifies the test to only pull the specific OpInfo and the index of the sample input
@example(opinfo="torch.add", sample_input_index=3)
@ops(op_db)
def test_vmap(self, device, dtype, op):
    for sample_input in op.sample_inputs():
        run_test(sample_input)
```

Reply:
Awesome, I like that `@example` syntax! I think something like that should be doable; lemme see what I can throw together.